src/figuresClasses.ts

```diff
@@ -1,19 +1,81 @@
-export interface Figure {
+export enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+export enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue,'
 }
 
-export class Triangle {
+export interface Figure {
+  color: Color;
+  shape: Shape;
 
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(public color: Color, public a: number,
+    public b: number, public c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides have to be higher than 0');
+    }
+
+    if (a + b <= c
+      || b + c <= a
+      || a + c <= b) {
+      throw new Error(`Sides ${a}, ${b} and ${c} can't form a traingle`);
+    }
+
+    this.a = a;
+    this.b = b;
+    this.c = c;
```

Comment:
In TypeScript, when you declare a constructor parameter with an access modifier TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property.
So in your code, public a: number in the constructor is equivalent to writing this.a = a. You can safely remove this.